src/Microsoft.DotNet.Interactive.Tests/LanguageKernelTests.cs

```diff
@@ -1264,5 +1267,208 @@ public async Task SetVariableAsync_can_redeclare_an_existing_variable_and_change
+        [InlineData(Language.FSharp, @"
+System.Threading.Thread.Sleep(3000)
+Console.WriteLine(""done c#"")", "done f#")]
+        public async Task user_code_can_react_to_cancel_command_using_cancellation_token(Language language, string code, string expectedValue)
+        {
+            var kernel = CreateKernel(language);
+            var cancelCommand = new Cancel();
+            var submitCodeCommand = new SubmitCode(code);
+
+            Task.Run(async () => await kernel.SendAsync(submitCodeCommand));
+            await Task.Delay(100);
+            await kernel.SendAsync(cancelCommand);
+
+
+            KernelEvents
+                .Should()
+                .ContainSingle<StandardOutputValueProduced>()
+                .Which
+                .Value
+                .Should()
+                .Be(expectedValue);
+
+        }
+
+        [Theory]
+        [InlineData(Language.CSharp)]
+        [InlineData(Language.FSharp)]
+        [InlineData(Language.PowerShell, Skip = "to address later")]
+        public void cancel_command_cancels_the_running_command(Language language)
+        {
+            var submitCodeCommandCancelled = false;
+            var submitCodeCommandStarted = false;
+            var kernel = CreateKernel(language);
+
+            var cancelCommand = new Cancel();
+
+            var submitCodeCommand = new SubmitCode("placeholder")
+            {
+                Handler = (command, context) =>
+                {
+                    submitCodeCommandStarted = true;
+
+                    while (!context.CancellationToken.IsCancellationRequested)
+                    {
+
+                    }
+
+                    submitCodeCommandCancelled = true;
+
+                    return Task.CompletedTask;
+                }
+            };
+
+            Task.WhenAll(
+                    Task.Run(async () =>
+                    {
+                        await kernel.SendAsync(submitCodeCommand);
+                    }),
+                    Task.Run(async () =>
+                    {
+                        await Task.Delay(100);
+                        await kernel.SendAsync(cancelCommand);
+                    })).Wait();
+              //  .Wait(TimeSpan.FromSeconds(20));
+
+
+            using var _ = new AssertionScope();
+
+            submitCodeCommandStarted.Should().BeTrue();
+            submitCodeCommandCancelled.Should().BeTrue();
+
+            KernelEvents
+                .Should()
+                .ContainSingle<CommandFailed>(c => c.Command == submitCodeCommand)
+                .Which
+                .Exception
+                .Should()
+                .BeOfType<OperationCanceledException>();
+        }
+
+        [Theory]
+        [InlineData(Language.CSharp)]
+        [InlineData(Language.FSharp)]
+        public async Task cancel_command_cancels_all_deferred_commands(Language language)
```

Comment:
This looks like a duplicate. Consolidating will help make it clearer what we're testing.